feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+import { BREAKPOINT, BreakpointService } from '@spartacus/storefront';
+import { take } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+export class ConfiguratorGroupMenuService {
+  constructor(
+    @Inject(PLATFORM_ID) protected platformId: any,
+    @Inject(DOCUMENT) protected document,
+    protected breakpointService?: BreakpointService
+  ) {}
+
+  protected getGroups(): NodeListOf<HTMLElement> {
+    if (isPlatformBrowser(this.platformId)) {
+      let selector = ' cx-configurator-group-menu button[role="tab"]';
+      this.breakpointService
+        ?.isUp(BREAKPOINT.lg)
+        .pipe(take(1))
+        .subscribe((isDesktop) => {
+          if (isDesktop) {
+            selector = 'main' + selector;
+          } else {
+            selector = '.navigation' + selector;
+          }
+        });
+      return document.querySelectorAll(selector);
+    }
+  }
+
+  protected getFocusedGroupIndex(groups): number | undefined {
+    if (isPlatformBrowser(this.platformId)) {
+      let focusedElement = document.activeElement;
+      let focusedElementId = focusedElement.id;
+      if (groups) {
+        for (let index = 0; index < groups.length; index++) {
+          if (groups[index].id === focusedElementId) {
+            return index;
+          }
+        }
+      }
+      return undefined;
+    }
+  }
+
+  protected updateCurrentGroupIndex(
+    currentGroupIndex: number,
+    focusedGroupIndex: number
+  ): number {
+    return focusedGroupIndex !== currentGroupIndex
+      ? focusedGroupIndex
+      : currentGroupIndex;
+  }
```

Comment:
This method doesn't update the current group index. But that's what I understand on the first sight from the method name.

Could you please change the name to be more descriptive? You might also consider to add a JSDoc comment for the method.

Reply:
If the current index ID is not equal focused index ID then the current index ID will be updated / changed to the focused index ID, otherwise the current index ID stays unchanged.
I thought, the protected methods should not get any JSDoc, please correct me

Reply:
I added the JSDocs to all methods in the service